SpeechTranslator/MainPage.xaml.cs

```diff
@@ -0,0 +1,513 @@
+        public StreamSocket clientSocket;
+        private bool connected = false;
+
+        public static Language en_lang = new Language("en-US");
+        public static Language ch_lang = new Language("zh-CN");
+
+        public static Language InputLang=en_lang;
+        public static Language OutputLang = ch_lang;
+        public static string synLangStr = "zh";
+
+        private SpeechRecognizer speechRecognizer;
+        private SpeechSynthesizer synthesizer;
+
+        private static uint HResultPrivacyStatementDeclined = 0x80045509;
+
+        public MainPage()
+        {
+            this.InitializeComponent();
+            startListener();
+            InitRecogAndSyn();
+
+        }
+        private async void startListener()
+        {
+            if (String.IsNullOrEmpty(ConstantParam.SelfPort))
+            {
+                StatusText.Text = "Please provide a service port for listening.";
+                return;
+            }
+            listener.ConnectionReceived += OnConnection;
+
+            try
+            {
+                await listener.BindServiceNameAsync(ConstantParam.SelfPort);
+                StatusText.Text = "Listening on port: " + ConstantParam.SelfPort;
+                ReceivedText.Text = "";
+            }
+            catch (Exception exception)
+            {
+                if (SocketError.GetStatus(exception.HResult) == SocketErrorStatus.Unknown)
+                {
+                    throw;
+                }
+                StatusText.Text = "Start listening failed with error: " + exception.Message;
+            }
+        }
+
+        private async void OnConnection(StreamSocketListener sender, StreamSocketListenerConnectionReceivedEventArgs args)
+        {
+
+            DataReader reader = new DataReader(args.Socket.InputStream);
+            try
+            {
+                while (true)
+                {
+                    // Read first 4 bytes (length of the subsequent string).
+                    uint sizeFieldCount = await reader.LoadAsync(sizeof(uint));
+                    if (sizeFieldCount != sizeof(uint))
+                    {
+                        // The underlying socket was closed before we were able to read the whole data.
+                        return;
+                    }
+
+                    // Read the string.
+                    uint stringLength = reader.ReadUInt32();
+                    uint actualStringLength = await reader.LoadAsync(stringLength);
+                    if (stringLength != actualStringLength)
+                    {
+                        // The underlying socket was closed before we were able to read the whole data.
+                        return;
+                    }
+
+                    // Display the string.
+                    string text = reader.ReadString(actualStringLength);
+
+                    string InLang ="en";
+                    string outLang = "zh-CHS";
```

Comment:
consider making these member variables for the class? I noticed we have these strings as well as four Language objects which actually only point to two objects (en_lang and InputLang). Are they all necessary?